contracts/HATVaultsRegistry.sol

```diff
@@ -0,0 +1,563 @@
+    * @param _periods The number of vesting periods. Must be more than 0 and 
+    * less then the vesting duration.
+    */
+    function setHatVestingParams(uint256 _duration, uint256 _periods) external onlyOwner {
+        if (_duration >= 180 days) revert VestingDurationTooLong();
+        if (_periods == 0) revert VestingPeriodsCannotBeZero();
+        if (_duration < _periods) revert VestingDurationSmallerThanPeriods();
+        generalParameters.hatVestingDuration = _duration;
+        generalParameters.hatVestingPeriods = _periods;
+        emit SetHatVestingParams(_duration, _periods);
+    }
+
+    /**
+    * @notice Called by governance to set the timelock delay for setting the
+    * max bounty (the time between setPendingMaxBounty and setMaxBounty)
+    * @param _delay The time period for the delay. Must be at least 2 days.
+    */
+    function setMaxBountyDelay(uint256 _delay)
+    external
+    onlyOwner {
+        if (_delay < 2 days) revert DelayTooShort();
+        generalParameters.setMaxBountyDelay = _delay;
+        emit SetMaxBountyDelay(_delay);
+    }
+
+    /**
+    * @notice Create a new vault
+    * @param _asset The vault's native token
+    * @param _committee The address of the vault's committee 
+    * @param _rewardController The reward controller for the vault
+    * @param _maxBounty The maximum percentage of the vault that can be paid
+    * out as a bounty. Must be between 0 and `HUNDRED_PERCENT`
+    * @param _bountySplit The way to split the bounty between the hacker, 
+    * committee and governance.
+    *   Each entry is a number between 0 and `HUNDRED_PERCENT`.
+    *   Total splits should be equal to `HUNDRED_PERCENT`.
+    *   Bounty larger than 0 must be specified for the hacker (direct or 
+    *   vested in vault's native token).
+    * @param _descriptionHash Hash of the vault description.
+    * @param _bountyVestingParams Vesting params for the bounty
+    *        _bountyVestingParams[0] - vesting duration
+    *        _bountyVestingParams[1] - vesting periods
+    * @param _isPaused Whether to initialize the vault with deposits disabled
+    * @return vault The address of the new vault
+    */
+    function createVault(
+        IERC20 _asset,
+        address _committee,
+        IRewardController _rewardController,
+        uint256 _maxBounty,
+        HATVault.BountySplit memory _bountySplit,
+        string memory _descriptionHash,
+        uint256[2] memory _bountyVestingParams,
+        bool _isPaused
+    ) 
+    external 
+    returns(address vault)
+    {
+        vault = Clones.clone(hatVaultImplementation);
+
+        HATVault(vault).initialize(
+            _rewardController,
+            _bountyVestingParams[0],
+            _bountyVestingParams[1],
+            _maxBounty,
+            _bountySplit,
+            _asset,
+            _committee,
+            _isPaused
+        );
+
+        hatVaults.push(vault);
+
+        emit VaultCreated(
```

Comment:
also emit a `SetVaultDescription` event?

Reply:
the description hash is already in the event here so i don't see the need

Reply:
it is easier to index the description by just subscribing to `SetVaultDescription` events, instead of having to subscribe also to the `VaultCreated` event as you have to do now

Reply:
ok, and it's useful? given that it's just the hash and not the description?